Assets/Scripts/EOSStatsManager.cs

```diff
@@ -0,0 +1,176 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+//#define ENABLE_DEBUG_EOSSTATSMANAGER;
+
+using System.Collections.Generic;
+using Epic.OnlineServices.Stats;
+using System;
+using Epic.OnlineServices;
+using System.Diagnostics;
+
+namespace PlayEveryWare.EpicOnlineServices.Samples
+{
+    /// <summary>
+    /// Class <c>EOSStatsManager</c> is a simplified wrapper for EOS [Stats Interface](https://dev.epicgames.com/docs/services/en-US/Interfaces/Stats/index.html).
+    /// </summary>
+    public class EOSStatsManager : EOSOnConnectLogin, IEOSSubManager
+    {
+        /// <summary>
+        /// Cache of the map between the product user id and the stats for that user.
+        /// </summary>
+        private Dictionary<ProductUserId, List<Stat>> _productUserIdToStatsCache = new();
+
+        protected override void PostSuccessfulLogin()
+        {
+            QueryStatsForProductUserId(_productUserId, (ref OnQueryStatsCompleteCallbackInfo statsQueryData) =>
+            {
+                CacheStatsForProductUserId(_productUserId);
+            });
+        }
+
+        /// <summary>
+        /// Gets the StatsInterface for EOS
+        /// </summary>
+        /// <returns>The stats interface.</returns>
+        private StatsInterface GetEOSStatsInterface()
+        {
+            return EOSManager.Instance.GetEOSPlatformInterface().GetStatsInterface();
+        }
+
+        private void CacheStatsForProductUserId(ProductUserId productUserId)
+        {
+            var statInterface = GetEOSStatsInterface();
+            GetStatCountOptions countOptions = new()
+            {
+                TargetUserId = productUserId
+            };
+            uint statsCountForProductUserId = statInterface.GetStatsCount(ref countOptions);
+
+            List<Stat> collectedStats = new();
+            var copyStatsByIndexOptions = new CopyStatByIndexOptions
+            {
+                TargetUserId = productUserId,
+                StatIndex = 0
+            };
+
+            for (uint i = 0; i < statsCountForProductUserId; ++i)
+            {
+                copyStatsByIndexOptions.StatIndex = i;
+
+                Result copyStatResult = statInterface.CopyStatByIndex(ref copyStatsByIndexOptions, out Stat? stat);
```

Comment:
not a critique: Legitimately didn't know you could use the null coalescing thing on an out. Interesting! I guess it provides another way of knowing if the function got something.

Reply:
Yeah - super helpful! :)